cmake/OpenCVPCHSupport.cmake

```diff
@@ -60,11 +74,15 @@ MACRO(_PCH_GET_COMPILE_FLAGS _out_compile_flags)
         endif()
     ENDFOREACH(item)
 
-    GET_DIRECTORY_PROPERTY(_directory_flags DEFINITIONS)
```

Comment:
@ilya-lavrenov , cmake 3.3.2 shows warning at this and next line: https://cmake.org/cmake/help/v3.3/policy/CMP0059.html

I added other changes from master branch to check if it will build on other platforms. Please, let me know if this file breaks your builds - I will revert parts not necessary for the fix.